Romiguier/Voleurs/src/Coffre.ts

```diff
@@ -0,0 +1,56 @@
+import type { EtatCoffre } from "./EtatCoffre";
+
+type CoffreParams = {
+  id: number;
+  nombreDePositions: number;
+  nombreDeSymboles: number;
+};
+
+export class Coffre {
+  private _id: number;
+  private _nombreDeCombinaisonsRestantes: number = 0;
+  private _etat: EtatCoffre = "fermé";
+  constructor({ id, nombreDePositions, nombreDeSymboles }: CoffreParams) {
+    this._nombreDeCombinaisonsRestantes = Math.pow(
+      nombreDeSymboles,
+      nombreDePositions
+    );
+    this._id = id;
+  }
+
+  get id() {
+    return this._id;
+  }
+
+  get nombreDeCombinaisonsRestantes(): number {
+    return this._nombreDeCombinaisonsRestantes;
+  }
+
+  get estOuvert(): boolean {
+    return this._etat === "ouvert";
```

Comment:
The _etat depends on the _nombreDeCombinaisonsRestantes, I don't see any interest of having an attribute to store this information.  estOuvert() just need to check the _nombreDeCombinaisonsRestantes

Reply:
I have removed all states and use the remaining number of combinations.